components/calendar/src/ixdtf.rs

```diff
@@ -0,0 +1,296 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+use core::str::FromStr;
+
+use crate::{AnyCalendar, Date, DateTime, Iso, RangeError, Time};
+use ixdtf::parsers::records::IxdtfParseRecord;
+use ixdtf::parsers::IxdtfParser;
+use ixdtf::ParserError;
+
+/// An error returned from parsing an IXDTF string to an `icu_calendar` type.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum FromIxdtfError {
+    /// Syntax error in the IXDTF string.
+    Ixdtf(ParserError),
+    /// Value is out of range of the `icu_calendar` type.
+    Range(RangeError),
+    /// The IXDTF is missing fields required for parsing into the chosen type.
+    MissingFields,
+    /// The IXDTF specifies an unknown calendar.
+    UnknownCalendar,
+}
+
+impl From<RangeError> for FromIxdtfError {
+    fn from(value: RangeError) -> Self {
+        Self::Range(value)
+    }
+}
+
+impl From<ParserError> for FromIxdtfError {
+    fn from(value: ParserError) -> Self {
+        Self::Ixdtf(value)
+    }
+}
```

Comment:
I thought about that but they are unavoidably in the public API as being a variant of `FromIxdtfError` (now `FromStrError`)

Reply:
A few choices here:

1. Ignore it and hope that `ixdtf::ParserError` stays semver compatible
2. Log the error and return `Syntax` with no details in the variant
3. Make sure the `ixdtf` crate reaches stable by the time we release the next `icu_calendar` crate version

CC @Manishearth @nekevss

Reply:
Define "stable"? The crate works and has a reasonable API right now, no?

These impls simplify some code in this file (barely), which I don't think is worth the cost of them showing up in docs, regardless of how stable ixdtf is.

Reply:
"stable" meaning version 1.0 of the `ixdtf` crate.

I don't quite understand the harm in having From impls when they are truly trivial, wrapping the error with a specific enum variant.

From impls barely show up in docs because they are a standard library trait implementation.

Reply:
I don't understand that requirement, `fixed_decimal`, `litemap`, `tinystr`, `writeable`, are all crates that appear on our APIs and are pre-1.0.

Reply:
I realized I don't really like option 2, either, because adding a value to the `Syntax` variant is also itself a semver change.